modules/kubernetes-addons/velero/main.tf

```diff
@@ -0,0 +1,45 @@
+####
+# Add your custom resources here...
+# such as IRSA policy statement
+####
+
+# Invokes the generic helm-addon module which is a convenience module
+# EKS Blueprints framework provides to create helm based addons easily
+
+
+module "helm_addon" {
+  source            = "github.com/aws-samples/aws-eks-accelerator-for-terraform//modules/kubernetes-addons/helm-addon"
+  manage_via_gitops = var.manage_via_gitops
+  set_values        = local.set_values
+  helm_config       = local.helm_config
+  irsa_config       = local.irsa_config
+  addon_context     = var.addon_context
+   depends_on = [
+    aws_s3_bucket.s3, 
+  ]
+}
+
+
+resource "aws_s3_bucket" "s3" {
+   bucket_prefix = local.s3bucketprefix
+  
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "encryption" {
+  bucket = aws_s3_bucket.s3.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm     = "AES256"
+    }
+  }
+}
```

Comment:
I would be more comfortable with customers providing us a `bucket_name` as variable or at least keep the bucket creation part optional.